plugins/search-react/src/components/HighlightedSearchResultText/HighlightedSearchResultText.tsx

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { makeStyles } from '@material-ui/core';
+
+const useStyles = makeStyles(
+  () => ({
+    highlight: {},
+  }),
+  { name: 'BackstageHighlightedSearchResultText' },
+);
+
+/**
+ * @public
+ */
+export type HighlightedSearchResultTextProps = {
+  text: string;
+  preTag: string;
+  postTag: string;
+};
+
+/**
+ * @public
+ */
+export const HighlightedSearchResultText = ({
+  text,
+  preTag,
+  postTag,
+}: HighlightedSearchResultTextProps) => {
+  const classes = useStyles();
+  const terms = text.split(new RegExp(`(${preTag}.+?${postTag})`));
```

Comment:
It _might_ be worthwhile to `useMemo` this, depending on how much preview is shown on a customized result page.

Also, it is worth noting that the actual content of the tag is only preserved because the regex is within a `(group)`.